examples/run_gate_benchmark.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+crt_dir=$(pwd)
+
+# Export env variables in case cmake reverts to default values
+export CXX=$(which clang++)
```

Comment:
Maybe defining a bash array with the expected compilers above here, and iterating over them, producing a build/output per compiler queried.

Reply:
Don't believe I fully understand the suggestion. Is the idea to test and compare multiple compilers by listing them as command-line arguments? E.g. `bash run_gate_benchmark.sh clang++ g++` would benchmark with both clang++ and g++?
If the idea is to iterate through possible installed compilers to find one that works then I believe the solution of passing the compiler name via a command-line argument circumvents this problem.

Reply:
Sure, passing the compiler explicitly as an argument would work. In that case, ignore my comment about having a list of available compilers in the file.